File: Nextech.Back.Api/Nextech.Model/DTO/ItemDTO.cs

using Nextech.Core.DTO.Enums;

namespace Nextech.Core.DTO;

public class ItemDTO
{
    public int ID { get; set; }
    public string Title { get; set; }
    public eItemType Type { get; set; }
    public DateTime Date { get; set; }
    public string Text { get; set; }
    public string Url { get; set; }
    public string By { get; set; }


    public bool IsDeleted { get; set; } 
    
    public bool IsDead { get; set; } 

    public string Parent { get; set; }

    public string Poll { get; set; } 

    public int[] Kids { get; set; } 

    public int Score { get; set; } 

    public int[] Parts { get; set; } // A list of related pollopts, in display order.

    public string Descendants { get; set; } // In the case of stories or polls, the total comment count.
}

File: Nextech.Web/src/api/model/item-dto.ts
import { eTypeItem } from "./enum/eType-Item"

export interface ItemDTO
{
    id: number  //The item's unique id.
    deleted: boolean    //true if the item is deleted.
    type: eTypeItem  // The type of item.One of "job", "story", "comment", "poll", or "pollopt".
    by: string  // The username of the item's author.
    date: number  // Creation date of the item, in Unix Time.
    text: string    // The comment, story or poll text. HTML.
    dead : boolean    // true if the item is dead.
    parent: string  // The comment's parent: either another comment or the relevant story.
    poll: string;  // The pollopt's associated poll.
    kids: number[];  // The ids of the item's comments, in ranked display order.
    url: string;  // The URL of the story.
    score: number;  // The story's score, or the votes for a pollopt.
    title: string;  // The title of the story, poll or job.HTML.
    parts: number[];  // A list of related pollopts, in display order.
    descendants: string;  // In the case of stories or polls, the total comment count.
}

